Fix WebFinger rel ID to match server implementation

michaelstingl opened the pull request:
Fixes #197

The desktop client was expecting `http://webfinger.opencloud.eu/rel/server-instance` but the server uses `http://webfinger.opencloud/rel/server-instance` (without .eu).

This change updates the desktop client to use the correct relation ID. Related to opencloud-eu/opencloud#65.

TheOneRing commented:
I Michael, thanks for the prompt fix. Could you please also update the comment in https://github.com/opencloud-eu/desktop/blob/main/src/libsync/creds/webfinger.cpp#L40-L40

Could you please also mention some of your findings in the https://github.com/opencloud-eu/desktop/blob/main/src/libsync/creds/webfinger.h#L39-L39
So where on the server this is defined?

Thank you

michaelstingl commented:
@TheOneRing I've updated the PR as requested:

1. Fixed the comment in `webfinger.cpp` to use the correct WebFinger rel ID.
2. Added additional information in `webfinger.h`:
   - Reference to the server implementation in `services/webfinger/pkg/relations/owncloud_instance.go`
   - References to the relevant RFCs (RFC 7033 WebFinger and RFC 5988 Web Linking)
   - Explanation about using the URI format for custom relation types

I've also rebased the branch to the latest main branch.
